src/marqo/s2_inference/clip_utils.py

```diff
@@ -182,15 +186,30 @@ def download_image_from_url(image_path: str, media_download_headers: dict, timeo
     headers.update(media_download_headers)
     c.setopt(pycurl.HTTPHEADER, [f"{k}: {v}" for k, v in headers.items()])
 
+    # callback to check file size for video and audio
```

Comment:
We could, and this was an option, but this would add latency since we're adding another request for each query. I figure a majority of usecases won't exceed 370 mb and it might not be worth adding the extra latency for only a small number of eventualities where file size is > 370 MB.